.github/workflows/main.yml

```
@@ -80,10 +80,22 @@ jobs:
         cd nemsio
         mkdir build 
         cd build
-        cmake .. -DENABLE_TESTS=ON -DCMAKE_PREFIX_PATH="~/pfunit;~/"
+        cmake .. -DENABLE_TESTS=ON -DCMAKE_PREFIX_PATH="~/pfunit;~/" -DCMAKE_Fortran_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0"
         make -j2
     
     - name: test
       run: |
         cd $GITHUB_WORKSPACE/nemsio/build
         make test
+
+    - name: run-gcovr
+      run: |
+        cd $GITHUB_WORKSPACE/nemsio/build
+        gcovr -r .. -v  --html-details -o test-coverage.html
+
+    - name: upload-test-coverage
+      uses: actions/upload-artifact@v2
+      with:
+        name: test-coverage
```

Small detail, but I like to have this be `<name>-test-coverage` because I end up with multiple `test-coverage.zip` files and don't know where they came from.

```suggestion
        name: nemsio-test-coverage
```